src/features/user/MFV2/ContributionsMap/Markers/index.tsx

```diff
@@ -69,15 +74,57 @@ const Markers = ({
         superclusterResponseForRegisteredTree
       );
     }
-  }, [viewport, registrationGeojson]);
+  }, [viewState, registrationGeojson]);
+
+  // Centers the map after the initial data is loaded
+  useEffect(() => {
+    if (!isCenteredOnce) {
+      const combinedSuperclusterResponse = [
+        ...donationSuperclusterResponse,
+        ...registrationSuperclusterResponse,
+      ];
+
+      if (combinedSuperclusterResponse.length === 0) {
+        return;
+      }
+
+      const markersGeojson = {
+        type: 'FeatureCollection',
+        features: combinedSuperclusterResponse.filter(
+          (feature) =>
+            feature.geometry &&
+            feature.geometry.type === 'Point' &&
+            Array.isArray(feature.geometry.coordinates) &&
+            feature.geometry.coordinates.length === 2
+        ),
+      };
+
```

Comment:
you can use `useMemo `to memoize the `combined supercluster response` and `the geojson object,`